src/lib.rs

```diff
@@ -241,6 +241,11 @@ pub fn crate_name(orig_name: &str) -> Result<FoundCrate, Error> {
 }
 
 fn workspace_manifest_path(cargo_toml_manifest: &Path) -> Result<Option<PathBuf>, Error> {
+    // Skip invoking Cargo if we're building a crate packaged with cargo.
+    if !cargo_toml_manifest.with_extension(".toml.orig").exists() {
```

Comment:
```suggestion
    if cargo_toml_manifest.with_extension(".toml.orig").exists() {
```

Isn't that the correct way?

Reply:
Because this file exists if the package was packaged by cargo, but also generally a little bit fragile because anything could create this file.

Maybe to fix your issue, just returning `None` when `CARGO` env variable wasn't found?

Reply:
Ooops, patch tested and submitted differ :( my bad, sorry.

Yes, I’d be happy to change the PR to return None of cargo isn’t set. I guess the corresponding error type can also go away then?

Reply:
To clarify: intended was this:

```rust
if cargo_toml_manifest.with_extension("toml.orig").exists() {
```

(note the removed leading dot)

Reply:
Pushed the version that I intended and that I verified that it works as intended.

I think it's a win-win kind of, as it also speeds up the regular builds. But if you prefer the None on unset Cargo env, then I can also do that.

Reply:
I'm a little bit afraid it breaks something else :D Also in a non obvious way. I think for now I would opt-in to check for `CARGO` being unchecked. But we could open an issue to speed up regular builds.

Reply:
Alright, sounds good. Force-push coming up to correct this :)